pkg/fqdn/cache.go

```diff
@@ -202,6 +252,34 @@ func (c *DNSCache) removeExpired(entries ipEntries, now time.Time) {
 	for ip, entry := range entries {
 		if entry == nil || entry.isExpiredBy(now) {
 			delete(entries, ip)
+			c.removeReverse(ip, entry)
 		}
 	}
 }
+
+// upsertReverse updates the reverse DNS cache for ip with entry, if it expires
+// later than the already-stored entry.
+// It is assumed that entry includes ip.
+func (c *DNSCache) upsertReverse(ip string, entry *cacheEntry) {
+	entries, exists := c.reverse[ip]
```

Comment:
`c.RLock`?

Reply:
Both the `*Reverse` functions are called from places that already hold the lock, no? Did I miss some path?

Reply:
The caller has the lock requirement in the description but not this function itself which makes this not obvious. Just add a comment to the function description

Reply:
Sorry, you're right.